### PR TITLE
build: add some usage examples to make help

### PR DESCRIPTION
**Pull request by jordanlewis**
Output:

```
Typical usage:
   make test                                     Run all unit tests.
   make test PKG=./pkg/sql                       Run all unit tests in the ./pkg/sql package
   make test PKG=./pkg/sql TESTS=TestParse       Run the TestParse test in the ./pkg/sql package.
   make bench PKG=./pkg/sql/parser BENCHES=BenchmarkParse   Run the BenchmarkParse benchmark in the ./pkg/sql/parser package.
   make testlogic                                Run all SQL Logic Tests.
   make testlogic FILES=prepare                  Run the logic test with filename prepare.
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,6 +78,18 @@ help: ## Print help for targets with comments.
 	@echo ""
 	@echo "Useful variables:"
 	@grep -Eh '^[a-zA-Z._-]+ *:=.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":=.*?## "}; {printf "  $(shell tput setaf 6 2>/dev/null)%-30s$(shell tput sgr0 2>/dev/null) %s\n", $$1, $$2}'
+	@echo ""
+	@echo "Typical usage:"
+	@grep -Eh '^## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = "##"}; {printf "  $(shell tput setaf 6 2>/dev/null)%-45s$(shell tput sgr0 2>/dev/null) %s\n", $$2, $$3}'
+
+#@grep -Eh '^## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = "##"}; {printf "  %s\n  $(shell tput setaf 6 2>/dev/null)$ %s$(shell tput sgr0 2>/dev/null)\n", $$3, $$2}'
```
Comment (benesch) [inline Makefile]: This looks stray?

File: Makefile [hunk 2]
```diff
@@ -238,7 +259,9 @@ bench: ## Run benchmarks.
 bench: TESTTIMEOUT := $(BENCHTIMEOUT)
 
 .PHONY: check test testshort testrace testlogic bench
+test: ## Run tests.
 check test testshort testrace bench: gotestdashi
+	$(warning You specified TESTS=$(TESTS) without specifying PKG - your tests will run against all packages)
```
Comment (benesch) [inline Makefile]: Holdover from an earlier attempt? This doesn't seem to check if `PKG` is actually empty.

File: Makefile [hunk 3]
```diff
@@ -22,6 +22,15 @@
 # `help` target. They look a bit awkward in the variable declarations below
 # since any whitespace added would become part of the variable's default value.
 
+ifeq ($(PKG),)
+ifneq ($(TESTS),)
+$(error TESTS must be specified with PKG (e.g. PKG=./pkg/sql/))
+endif
+ifneq ($(BENCHES),)
+$(error BENCHES must be specified with PKG (e.g. PKG=./pkg/sql/))
+endif
```
Comment (benesch) [inline Makefile]: This isn't strictly true, right? `make test TESTS=TestFoo` will run `TestFoo` in every package in which `TestFoo` exists, which is perhaps desirable. Can we perhaps downgrade these to `$(warning...)`s?

Comment (benesch) [reply Makefile]: Oh, just saw your comment about `PKG = ./pkg/...`. SGTM.

File: Makefile (hunk 1 above)
Comment (jordanlewis) [reply Makefile]: Done, ty

**Review by benesch — APPROVED** [review]

**Pull request merged**